doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
-whenever an error is raised or a failure happens, respectively. Inside the
+this, ``pytest`` can drop into a debugger whenever a test fails or raises an
+exception. This can be accomplished by passing ``--pdb``
+to the ``pytest`` executable. Inside the
 debugger you can interactively print out variables and go up and down the call
 stack to determine the context for your failure or error.
 
 .. code-block:: bash
 
-    nosetests --pdb --pdb-failures
+    pytest --pdb
 
 In addition, one can debug more crudely using print statements. To do this,
 you can add print statements to the code as normal. However, the test runner
 will capture all print output by default. To ensure that output gets printed
-to your terminal while the tests are running, pass ``-s`` to the ``nosetests``
+to your terminal while the tests are running, pass ``-s`` to the ``pytest``
 executable.
 
+.. code-block:: bash
+
+    pytest -s
+
 Lastly, to quickly debug a specific failing test, it is best to only run that
 one test during your testing session. This can be accomplished by explicitly
-passing the name of the test function or class to ``nosetests``, as in the
+passing the name of the test function or class to ``pytest``, as in the
 following example:
 
 .. code-block:: bash
 
-    $ nosetests yt.visualization.tests.test_plotwindow:TestSetWidth
+    $ pytest yt/visualization/tests/test_plotwindow.py::TestSetWidth
 
-This nosetests invocation will only run the tests defined by the
+This pytest invocation will only run the tests defined by the
 ``TestSetWidth`` class.
 
 Finally, to determine which test is failing while the tests are running, it helps
 to run the tests in "verbose" mode. This can be done by passing the ``-v`` option
-to the ``nosetests`` executable.
+to the ``pytest`` executable.
+
+.. code-block:: bash
 
-All of the above ``nosetests`` options can be combined. So, for example to run
+    $ pytest -v
+
+All of the above ``pytest`` options can be combined. So, for example, to run
 the ``TestSetWidth`` tests with verbose output, letting the output of print
 statements come out on the terminal prompt, and enabling pdb debugging on errors
 or test failures, one would do:
 
 .. code-block:: bash
 
-    $ nosetests --pdb --pdb-failures -v -s yt.visualization.tests.test_plotwindow:TestSetWidth
+    $ pytest --pdb -v -s yt/visualization/tests/test_plotwindow.py::TestSetWidth
+
+More pytest options can be found by using the ``-h`` option
+
+.. code-block:: bash
+
+    $ pytest -h
 
 .. _answer_testing:
 
 Answer Testing
 --------------
 
-What do Answer Tests Do
+What Do Answer Tests Do
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-Answer tests test **actual data**, and many operations on that data, to make
-sure that answers don't drift over time.  This is how we test frontends, as
-opposed to operations, in yt.
+Answer tests use `actual data <https://yt-project.org/data/>`_ to test reading, writing, and various manipulations of that data.
```

Comment:
Add back in the frontends sentence here, since most of our new users will end up needing to write answers for frontends and generally this is the first step for frontend testing.